apollo-federation/src/query_plan/query_planner.rs

```diff
@@ -348,6 +367,11 @@ impl QueryPlanner {
             NamedFragments::new(&document.fragments, &self.api_schema),
             &self.api_schema,
             &self.interface_types_with_interface_objects,
+            &|| {
+                QueryPlanningParameters::check_cancellation_with(
+                    &options.check_for_cooperative_cancellation,
+                )
```

Comment:
I was wondering about `check_cancellation_with` at first. Then, I realized `normalize_operation` has to be called before  `QueryPlanningParameters` is constructed. That's unfortunate.